tests/Serializer/Doctrine/ObjectConstructorTest.php

```diff
@@ -217,6 +221,19 @@ public function testNamingForIdentifierColumnIsConsidered()
         );
     }
 
+    public function testObjectManagerIsInjected(): void
+    {
+        $serializer = $this->createSerializerWithObjectManagerAwareFallback();
+        $jsonData = '{"name":"Hello Thing"}';
+        /** @var Thing $thing */
+        $thing = $serializer->deserialize($jsonData, Thing::class, 'json');
+
+        static::assertInstanceOf(ObjectManager::class, $thing->getObjectManager());
```

Comment:
can you modify the test to be able to use `assertSame` and check the exact instance?

Reply:
i dont use symfony at work but i will try. maybe this weekend.

Reply:
done